OnlineFlightSearchAPITestCases/ServiceTests/FlightServicesTests/FlightServiceTests.cs

```diff
@@ -0,0 +1,238 @@
+
+            var mockAirportService = new AirportServices(mockAirportRepository.Object);
+
+            var mockFlightRepo = new Mock<IFlightRepository>();
+            mockFlightRepo.Setup(x => x.flightDetails).Returns(GetFlightDetails());
+            mockFlightRepo.Setup(x => x.FetchFlightDetails(startDestination, endDestination, It.IsAny<DateTime>())).Returns(expectedFlightDetails().ToList());
+
+            ISearchFlightService searchFlightService = new FlightService(mockFlightRepo.Object, mockAirportService);
+
+            var actualResult = searchFlightService.SearchFlightDetails(startDestination, endDestination, DateTime.UtcNow.AddDays(1));
+
+            var expectedCount = expectedFlightDetails().Count;
+
+            Assert.IsType<List<FlightDetail>>(actualResult);
+            Assert.Equal(expectedCount, actualResult.Count);
+            Assert.True(actualResult.TrueForAll(x => x.StartLocation == startDestination));
+            Assert.True(actualResult.TrueForAll(x => x.Destination == endDestination));
+            Assert.True(actualResult.TrueForAll(x => x.DepartureDate.Date == (DateTime.UtcNow.AddDays(1).Date)));
+        }
+
+        [Theory]
+        [InlineData("BUD", "IAD")]
+        public void SearchFlight_IfNoMatchFoundForFlightSearch_ThrowsValidationException(string startLocation, string endLocation)
+        {
+            var mockAirportRepository = new Mock<IAirportRepository>();
+            mockAirportRepository.Setup(x => x.airportDetails).Returns(GetAirportDetails());
+
+            var mockAirportService = new AirportServices(mockAirportRepository.Object);
+
+            var mockFlightRepo = new Mock<IFlightRepository>();
+            mockFlightRepo.Setup(x => x.flightDetails).Returns(GetFlightDetails());
+            mockFlightRepo.Setup(x => x.FetchFlightDetails(startLocation, endLocation, It.IsAny<DateTime>())).Throws<ValidationException>();
+
+            ISearchFlightService searchFlightService = new FlightService(mockFlightRepo.Object, mockAirportService);
+
+            var expectedException = new ValidationException();
+            var actualException = Assert.Throws<ValidationException>(() => searchFlightService.SearchFlightDetails(startLocation, endLocation, DateTime.UtcNow.AddDays(1)));
+
+            Assert.Equal(expectedException.Message, actualException.Message);
+        }
+
+        private List<FlightDetail> expectedFlightDetails()
+        {
+            var fixture = new Fixture();
+            List<FlightDetail> flightDetails = fixture.Build<FlightDetail>()
+                             .With(x => x.StartLocation, "BUD")
+                             .With(x => x.Destination, "LTN")
+                             .With(x => x.DepartureDate, DateTime.UtcNow.AddDays(1))
+                             .With(x => x.TravelTime, "2:35").CreateMany(4).ToList();
+
+            return flightDetails;
+        }
+
+
+        private List<FlightDetail> GetFlightDetails()
+        {
```

Comment:
This seems excessively hard coded. Can you try reducing it? May be pick random from list of predefied values?

Reply:
Predefined a list and fetched from the same.(To satisfy valid test case scenario, some are hardcoded)